stdlib/jsonutils/src/main/java/org/ballerinalang/stdlib/jsonutils/XmlToJsonConverter.java

```diff
@@ -0,0 +1,362 @@
+        BXML next = iterator.next();
+        if (next.getNodeType() == XMLNodeType.TEXT) {
+            return SequenceConvertibility.LIST;
+        }
+        // Scan until first convertible item is found.
+        while (iterator.hasNext() && (isCommentOrPi(next))) {
+            next = iterator.next();
+            if (next.getNodeType() == XMLNodeType.TEXT) {
+                return SequenceConvertibility.LIST;
+            }
+        }
+
+        String firstElementName = next.getElementName();
+        int i = 0;
+        boolean sameElementName = true;
+        for (; iterator.hasNext(); i++) {
+            BXML val = iterator.next();
+            if (val.getNodeType() == XMLNodeType.ELEMENT) {
+                if (!((XMLItem) val).getElementName().equals(firstElementName)) {
+                    sameElementName = false;
+                }
+            } else if (val.getNodeType() == XMLNodeType.TEXT) {
+                return SequenceConvertibility.LIST;
+            } else {
+                i--; // we don't want `i` to count up for comments and PI items
+            }
+        }
+        return (sameElementName && i > 0) ? SequenceConvertibility.SAME_KEY : SequenceConvertibility.ELEMENT_ONLY;
+    }
+
+    private static ArrayValueImpl newJsonList() {
+        return new ArrayValueImpl(new BArrayType(BTypes.typeJSON));
+    }
+
+    public static ArrayValueImpl newJsonListFrom(List<Object> items) {
+        return new ArrayValueImpl(items.toArray(), new BArrayType(BTypes.typeJSON));
+    }
+
+    private static MapValueImpl<BString, Object> newJsonMap() {
+        return new MapValueImpl<>(jsonMapType);
+    }
+
+    /**
+     * Extract attributes and namespaces from the XML element.
+     *
+     * @param element XML element to extract attributes and namespaces
+     * @param preserveNamespaces should namespace attribute be preserved
+     */
+    private static LinkedHashMap<String, String> collectAttributesAndNamespaces(XMLItem element,
+                                                                                boolean preserveNamespaces) {
+        int nsPrefixBeginIndex = XMLItem.XMLNS_URL_PREFIX.length();
+        LinkedHashMap<String, String> attributeMap = new LinkedHashMap<>();
+        Map<String, String> nsPrefixMap = new HashMap<>();
+        for (Map.Entry<BString, BString> entry : element.getAttributesMap().entrySet()) {
+            if (entry.getKey().getValue().startsWith(XMLItem.XMLNS_URL_PREFIX)) {
+                String prefix = entry.getKey().getValue().substring(nsPrefixBeginIndex);
+                String ns = entry.getValue().getValue();
+                nsPrefixMap.put(ns, prefix);
+                if (preserveNamespaces) {
+                    if (prefix.equals(XMLNS)) {
+                        attributeMap.put(prefix, ns);
+                    } else {
+                        attributeMap.put(XML_NAMESPACE_PREFIX_FRAGMENT + prefix, ns);
+                    }
+                }
+            }
```

Comment:
Shall we change this to something like

```suggestion
            if (!entry.getKey().getValue().startsWith(XMLItem.XMLNS_URL_PREFIX)) {
                continue;
           }
           String prefix = entry.getKey().getValue().substring(nsPrefixBeginIndex);
           String ns = entry.getValue().getValue();
           nsPrefixMap.put(ns, prefix);
           if (preserveNamespaces) {
               if (prefix.equals(XMLNS)) {
                   attributeMap.put(prefix, ns);
               } else {
                   attributeMap.put(XML_NAMESPACE_PREFIX_FRAGMENT + prefix, ns);
               }
            }
```